rust/benches/scan.rs

```diff
@@ -31,7 +31,9 @@ fn bench_scan(c: &mut Criterion) {
         b.to_async(&rt).iter(|| async {
             let count = dataset
                 .scan()
-                .into_stream()
+                .try_into_stream()
```

Comment:
It'd be awesome if we can catch such error during CI.

Reply:
it is odd, CI catched my other PR error where I don't have this change https://github.com/eto-ai/lance/actions/runs/4025972335/jobs/6922761254